seller page contains top level formatting

RyanWang3 opened the pull request:
- created front end skeleton for sellers home
- not connected to API yet

zhou-michael committed:
initial getListing implementation

zhou-michael committed:
error handling

zhou-michael committed:
patch listing implemented

zhou-michael committed:
error message edit

zhou-michael committed:
fix try catch on patch

zhou-michael committed:
removed loop in updateDoc, bugfixes

zhou-michael committed:
timestamp

zhou-michael committed:
spacing

zhou-michael committed:
addListing implemented

RyanWang3 committed:
seller page contains top level formatting

zhou-michael committed:
snapshot user info in new listing

JonathonYY commented:
Include the required modules (reviewing the pr locally gives `Module not found: Can't resolve '@mui/material'` even after `npm install`)

RyanWang3 committed:
add package.json

JonathonYY commented:
Looks good for now - maybe begin mocking the separate display for `selected_buyer` while waiting for APIs to merge?

RyanWang3 committed:
finalized initial my listings page

JonathonYY commented:
Your latest commit also requires `@mui/icons-material`, so `package.json` should be updated again.

Couple questions:
![Screenshot 2025-02-24 at 10 10 33 PM](https://github.com/user-attachments/assets/a78dd010-403a-4d8b-867a-3a97b635d4ac)
Is the text supposed to look like this? It's kind of hard to read.

Is it possible for the "New Listing" button to be bound to the bottom of the page, instead of floating depending on how many products are owned? Additionally, I noticed when testing that when the list of products is very long, it extends the page - would it be possible to have the product list scroll instead?

RyanWang3 commented:
I looked into it and it seems that there's some inconsistency between the global css and this file's style. I'll look into a fix. I'll also see if I can implement a scroll for the your products.

zhou-michael committed:
begin unit tests, fix API problems

zhou-michael committed:
bugfixes

RyanWang3 committed:
implemented scrolling on left and right panels and add listing fixed to bottom

KevZ1209 committed:
initial image uploads -- frontend

zhou-michael committed:
fix up POST listing and add unit tests

KevZ1209 committed:
separate client and server side versions. use nextJS Image tag instead of img

KevZ1209 committed:
add typescript types

zhou-michael committed:
test cases for API calls, small fixes

zhou-michael committed:
complete rebase

zhou-michael committed:
style fixes

zhou-michael committed:
style changes, lower case in patch, fixed seller_rating

zhou-michael committed:
separate describe for PATCH and GET api calls

zhou-michael committed:
tests for div 0

KevZ1209 committed:
add image API and example page.tsx code that use it

KevZ1209 committed:
rename server-side to api-version

zhou-michael committed:
patch doesn't return reporter/ratings, small style fixes, use of arrayUnion

KevZ1209 committed:
remove log statements

KevZ1209 committed:
add image api test

KevZ1209 committed:
remove example image upload pagess

KevZ1209 committed:
update comment

zhou-michael committed:
remove log statements

RyanWang3 committed:
fixed scrolling issue in right panel

zhou-michael committed:
change default rating to 3.5, style fixes

zhou-michael committed:
nonnegative price

zhou-michael committed:
Listing API (#7)

* initial getListing implementation

* error handling

* patch listing implemented

* error message edit

* fix try catch on patch

* removed loop in updateDoc, bugfixes

* timestamp

* spacing

* addListing implemented

* snapshot user info in new listing

* begin unit tests, fix API problems

* bugfixes

* fix up POST listing and add unit tests

* test cases for API calls, small fixes

* complete rebase

* style fixes

* style changes, lower case in patch, fixed seller_rating

* separate describe for PATCH and GET api calls

* tests for div 0

* patch doesn't return reporter/ratings, small style fixes, use of arrayUnion

* remove log statements

* change default rating to 3.5, style fixes

* nonnegative price

eliu0120 commented on src/app/sellers_home/page.tsx:
```diff
@@ -1,36 +1,171 @@
+import icon from "@mui/icons-material/icon";
 
-            <p>Really cool listings stuff go here... this shit sounds like the hardest to impl ngl</p>
+const products = [
```
Make sure the listings parameter and users parameters matches those in the firebase

eliu0120 commented on src/app/sellers_home/page.tsx:
```diff
@@ -1,36 +1,171 @@
 
-            <hr />
+// import SideMenu from "@/components/seller_sidebar";
+import { Box } from "@mui/material";
```
this isn't used

eliu0120 commented on src/app/sellers_home/page.tsx:
```diff
@@ -1,36 +1,171 @@
+import StarIcon from "@mui/icons-material/Star";
+import AddIcon from "@mui/icons-material/Add";
+import React, { useState, useEffect } from "react";
+import icon from "@mui/icons-material/icon";
```
neither is this

eliu0120 reviewed (changes requested):
Left some comments

KevZ1209 committed:
add uuid prefix to image uploads. update unit tests

KevZ1209 committed:
update package-lock

KevZ1209 committed:
Kevinz oauth (#15)

* initial google signin/signout

* restrict to g.ucla.edu emails

* Listing APIs (#10)

* Implement rate API

* update rate to better use firebase

* initial report implementation

* Implement listing delete API

* Update timestamp on listing change

* Update interfaces

* Begin defining rate unit tests

* Complete rate API unit tests

* Split deleteListing logic to separate function

* Switch to log at warning level

* fix minor typo

* Unit tests for delete listing

* Update report API

Only allow valid users to report a listing
Enforce timeout between reports
Automatically delete listing once it has been reported a
certain number of times

* Unit tests for report listing

* Expand tests, input validation

* Implement User API (#8)

* post user api

* get user api

* patch and delete api

* remove manual test

* refactor firestore util

* testing framework and mock

* add phone number field

* remove phone number from post

* update firestore mocks

* remove id from db object

* unit tests

* add post unit test

* remove internal field from response

* more errors and testing

* test delete user

* Query listings (#12)

* Begin implementing API to get all listings with search

TODO: Expand search options
Create firestore index
Clarify API inputs

* Implement get all listings without search

Correct input handling to process values for pagination
Update query to only get listings without a match
Improve legibility

Note: Currently a call with a search string will not work

* Correct return structure to match spec

* Create helpers to parse input string

* Implement search querying

* Test getAllListings API

* Clarify query params optional

* add user details fetching for front page

* add auth context with tokens. update login and home pages to use the authcontext

* redirect logged in user to account page. update img to nextJS Image tags in home.tsx

* update login page with google login button

* update next.config.ts to allow google profile pictures

* remove debug buttons and text

---------

Co-authored-by: JonathonYY <43181115+JonathonYY@users.noreply.github.com>
Co-authored-by: Edwin He <41037314+Edwinhe03@users.noreply.github.com>

RyanWang3 committed:
removed uneeded imports and modified setting where product has no interested buyers

RyanWang3 committed:
API skeleton implemented

zhou-michael committed:
id returned with GET

RyanWang3 committed:
connected API to left panel

RyanWang3 committed:
changed key to id rather than title now that listings have id field

RyanWang3 committed:
implemented accept interested buyer

RyanWang3 committed:
added modify listing button, red x button now removes potential buyer

RyanWang3 committed:
displays buyer info once a buyer is selected

RyanWang3 committed:
implemented seller ratings

RyanWang3 committed:
seller page complete

RyanWang3 committed:
create listing button routes to the right page

Edwinhe03 commented on src/app/sellers_home/page.tsx:
```diff
@@ -1,36 +1,419 @@
+    }
+       
+    useEffect(() => {
+      console.log("bru",user_id);
```
extra log?

RyanWang3 replied:
fixed

RyanWang3 committed:
removed unnecessary logs

RyanWang3 committed:
redicts to login if user not logged in

RyanWang3 committed:
implemented loading page

RyanWang3 committed:
fixed bug where loading page wouldn't resolve

RyanWang3 committed:
added snackbar message
